src/main/java/messaging/MessageController.java

```diff
@@ -0,0 +1,24 @@
+package messaging;
+
+import models.MessageModel;
+
+public class MessageController {
+    private final MessageInputBoundary inputBoundary;
+
+
+    public MessageController(MessageInputBoundary inputBoundary){
+        this.inputBoundary = inputBoundary;
+    }
+    /**
+
+     */
+    public void handleSendMessage(Integer roomID, Integer sender, String content){
+        MessageModel request = new MessageModel(roomID, content, sender, false);
+        inputBoundary.handleSendMessage(request);
+    }
+
+    public void handleRetrieveMessages(Integer roomID, Integer sender, String content){
```

Comment:
seems like a violation of SRP? I think you should have each controller for each use case.

Reply:
I agree, I'll make a change